src/ws.c

```diff
@@ -144,7 +170,7 @@ int ws_sendframe(int fd, const char *msg, bool broadcast)
 		pthread_mutex_lock(&mutex);
 			for (i = 0; i < MAX_CLIENTS; i++)
 			{
-				sock = client_socks[i];
+				sock = client_socks[i].client_sock;
```

Comment:
Broadcasts (IMHO) should be restricted to clients connected through the same port, not for all clients. A quick loop could find the port for the broadcast sender and then you can check that port in the broadcast loop. =). If you disagree or have another thoughs on that, please let me know.

Reply:
Yes agree but in fact i am currently using wsServer in a global broadcast way (all clients receive the same page update/information). Maybe the "broadcast" parameter type should be changed to pass more options and support both broadcast modes (client broadcast, global broadcast) ?

Reply:
dismiss my previous answer, i have overlooked your answer. Indeed broadcasting between the ports has very little sense ;)
i will change this point.

Reply:
Changes done and tested !